hyperparameter_tuning/bayes_optuna/optuna_hpo.py

```diff
@@ -78,15 +98,15 @@ def __call__(self, trial):
         return actual_sla_value
 
 
-def recommend(application_name, direction, hpo_algo_impl, id, objective_function, tunables, value_type):
+def recommend(application_name, direction, hpo_algo_impl, id_, objective_function, tunables, value_type):
```

Comment:
`id` is the name of a [built-in function](https://docs.python.org/3/library/functions.html#id) in Python. 

In PEP 8 - Style Guide for Python Code, the following guidance appears in the section [Descriptive: Naming Styles](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles):

> single_trailing_underscore_: used by convention to avoid conflicts with Python keyword

Thus, `id` is changed to `id_` here.